src/gui/qgsmapcanvas.cpp

```diff
@@ -2621,7 +2624,7 @@ void QgsMapCanvas::wheelEvent( QWheelEvent *e )
 
 void QgsMapCanvas::setWheelFactor( double factor )
 {
-  mWheelZoomFactor = factor;
+  mWheelZoomFactor = std::max( factor, 1.01 );
```

Comment:
What is the reason for this?

Reply:
When you set a zoom factor < 1, clicking on the map canvas with the zoom in tool activated actually zooms out and vice versa. 1.01 is already the min value when you set it from the settings page, this only ensures that you cannot set it programmatically to less than 1